### DISCUSSION TITLE
How to get the tablet name using otd?

### DISCUSSION THREAD

**Discussion by ExWaltz** [OpenTabletDriver/OpenTabletDriver#2181] [Q&A] [answered]
I was using otd and I want to set output mode of my tablet.
In the help page it shows that you need to do `otd setoutputmode <tablet> <mode>` to set the output.
You can get `<mode>` with `otd listoutputmodes`. But there is no command in the help page that shows a way to get the `<tablet>`.

I don't want to use `otd-gui` because I want to make a script that lets me quickly change modes.

**Comment by vedxyz** [comment, marked as answer]
> I don't want to use otd-gui because I want to make a script that lets me quickly change modes

You might want to look into presets. They're supported in both otd.Console and otd.UX. So you would just need to save presets with the configurations you want. With the UX, you can quickly switch presets from the file menu, and on non-linux they're listed on the tray icon right click menu as well.

If you're set on going forward without presets:

https://github.com/OpenTabletDriver/OpenTabletDriver/blob/548d64d449fbfe391492cbacf2fdf33be5a681a5/OpenTabletDriver.Console/Program.Commands.cs#L218

The `getallsettings` command does give you a somewhat trivial to parse tablet name. I don't think there's a convenient command to give you just the name of a tablet, but you could also hardcode the name of your tablet into your script if that's sufficient for your use case. Note that the code above is run for all tablets detected, so that's something to consider if you've multiple.

That being said, it could be of interest to otd to have a query command for easily parsable outputs.
